### ISSUE TITLE
UK Channel STV - stream stalled

### ISSUE THREAD

**Issue by tibbsbrookside** [Catch-up-TV-and-More/plugin.video.catchuptvandmore#978]
### Describe the bug
When watching any STV catch up programmes, the stream stalls every few minutes.

### To Reproduce
Steps to reproduce the behaviour:
1. Play any STV catch up programme.

### Expected behaviour
Programme plays without interruption.

### Actual behaviour
Stream stalls every few minutes.

### Screenshots
None.

### Your environment
 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [x] Other

 - Operating system version/name: LibreELEC 10.0.3
 - Kodi version: 19.4
 - Catch-up TV & More version: 0.2.39~beta10
 - Installation method: https://catch-up-tv-and-more.github.io/installation/

### Additional context
Problem occurs on multiple LibreELEC systems, that have the same configuration, on the same local network.
Problem occurs even at the lowest resolution.
Problem occurs when connection is both direct and when using a VPN.
Internet connection is full fibre 150Mb/s.
iftop shows the peak throughput from Akamai is only around 2Mb/s (which I'm sure is not enough).
Streaming from other channels, such as channel 4 and UKTV Play, have no problems and I see peak throughput from CDNs such as CloudFlare at around 10Mb/s.
Other addons that use similar CDNs have no problem.

Example from the full debug log below:
2022-10-26 19:42:48.864 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:43:21.233 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:43:49.335 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:44:24.264 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:44:56.233 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:45:29.417 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:45:57.505 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:46:28.513 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:46:56.517 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:47:25.176 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:47:59.669 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:48:32.473 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:49:05.440 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:49:35.534 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled
2022-10-26 19:50:02.692 T:5823     INFO <general>: CVideoPlayerAudio::Process - stream stalled

I'm not sure if this is a problem with this addon or a generic network problem. To me it seems like a throughput problem from Akamai but I can't be sure (or what I can do if it is) and I've exhausted all the investigations I can think of. I'm hoping the debug log below will help provide an explanation.

Lastly, just to say thank you for this addon, I find it very useful.

### Debug log
https://paste.kodi.tv/juwacewebo.kodi

**Comment by joaopa00** [comment]
I tested emmerdale. No problem for me in Linux neither in AndroidTV.

**Comment by tibbsbrookside** [comment]
Thanks for the response.

I just tried Emmerdale and had the same problem.

Would you be able to confirm for me that the stream is coming from Akamai, and what the average download throughput is?

I'm sure this is a bandwidth problem but I can't work out why if I have the same problem when connected directly and via a VPN, unless Akamai are throttling traffic to both my ISP and my VPN provider.

**Comment by tibbsbrookside** [comment]
I've found the cause of my problems, and it has nothing to do with the addon, so apologies.

For info, the problem was the "no logs" public DNS provider I was using. I noticed in a network trace the amount of DNS calls that were being made when playing a programme, and the number where the response was NXDOMAIN. I've now switched to a different DNS provider and the throughput is up where I'd expect it to be and there are no "stream stalled".
